src/errors.rs

```diff
@@ -9,6 +9,16 @@ use std::str::Utf8Error;
 use std::string::FromUtf8Error;
 use std::sync::Arc;
 
+#[derive(Clone, Debug)]
+pub struct ReservedNamespacePrefixError {
+    pub prefix: String,
+}
```

Comment:
Is there a particular reason to make separate structs instead of directly embedding the strings in the enum variants?  I don't think we use this pattern in many, if any, places.

Reply:
I've collapsed them into a single struct. I think improving the errors should probably be left for another change. I hope this addresses your concern.